### PR TITLE
[NFI]: Cleanup TritonIntelGPUToLLVM codegen utilities

### PR DESCRIPTION
**Pull request by etiotto**
Remove unused functions, headers, etc...

### CODE REVIEW

File: third_party/intel/lib/TritonIntelGPUToLLVM/Utility.cpp [hunk 1]
```diff
@@ -67,19 +64,6 @@ static Value shuffleCommon(Location loc, RewriterBase &rewriter, Value val,
   return result;
 }
 
-Value loadShared(ConversionPatternRewriter &rewriter, Location loc, Value ptr,
```
Comment (etiotto) [inline third_party/intel/lib/TritonIntelGPUToLLVM/Utility.cpp:70]: Never referenced

File: third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h [hunk 2]
```diff
@@ -132,13 +125,13 @@ static Value getModuleWarpSize(RewriterBase &rewriter, Location loc) {
...
 // Shared memory utilities
 // -----------------------------------------------------------------------
-using ::mlir::triton::getMultiDimIndex;
-using ::mlir::triton::gpu::BlockedEncodingAttr;
-using ::mlir::triton::gpu::CTALayoutAttr;
-using ::mlir::triton::gpu::DotOperandEncodingAttr;
-using ::mlir::triton::gpu::NvidiaMmaEncodingAttr;
-using ::mlir::triton::gpu::SliceEncodingAttr;
-using ::mlir::triton::gpu::intel::DpasEncodingAttr;
+// using ::mlir::triton::getMultiDimIndex;
```
Comment (etiotto) [inline third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h:128]: Need to remove the commented out lines

File: third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h [hunk 3]
```diff
@@ -457,48 +450,6 @@ namespace mlir::triton::intel {
 inline SmallVector<SmallVector<unsigned>>
 emitOffsetForLayout(Attribute layout, RankedTensorType type);
 
-inline SmallVector<SmallVector<unsigned>>
-emitOffsetForSliceLayout(const SliceEncodingAttr &sliceLayout,
```
Comment (etiotto) [inline third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h:461]: Never referenced

File: third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h [hunk 4]
```diff
@@ -605,174 +556,11 @@ emitIndices(Location loc, RewriterBase &rewriter, const TargetInfoBase &target,
   return multiDimIdx;
 }
 
-/* ---------------- */
-/* ---------------- */
-inline DenseMap<unsigned, Value> getSwizzledSharedPtrs(
```
Comment (etiotto) [inline third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h:610]: Never referenced

**Review by anmyachev — APPROVED** [review]
LGTM!

**Pull request merged**